### ISSUE TITLE
Getting "Attempting to use uninitialized value" error when running base command

### ISSUE THREAD

**Issue by b6fung** [tf-coreml/tf-coreml#233]
So I have made my own pb file, and put in the correct layer name but when trying to run the command i run into the error Attempting to use uninitialized value fclayer/kernel,  when investigating that error it seems to be from initializing a session but the example in running it should just use one command.  

What could the error mean?

I also wish to note that the pb file that i saved was from tf.train.write_graph since i also seem to have trouble saving my simple model.  
Thanks

**Comment by DawerG** [comment]
You might have to initialize the variables in the session using     
```
init_op = tf.global_variables_initializer()
sess.run(init_op)
```

Nonetheless, @b6fung could you provide your code for inspection/reproduction?

**Comment by b6fung** [comment]
Thanks for the reply, this was the code i used to run the command

import tfcoreml as tf_converter

tf_converter.convert(tf_model_path = "./simple_as_binary.pb", mlmodel_path="./simple.mlmodel",output_feature_names=['import/fclayer/BiasAdd:0'])

I also used the pb inspect file to find my final layer.

**Comment by b6fung** [comment]
I can share my code, the data i cannot, but the model is very simple thank you very much for your time
`from __future__ import print_function
import numpy as np
import tensorflow as tf


# Parameters
learning_rate = 0.001
num_steps = 200
display_step = 25
batch_size = 5


# Network Parameters
n_hidden_1 = 256 # 1st layer number of neurons
n_hidden_2 = 1530 # 2nd layer number of neurons
n_hidden_3 = 1530
num_input = 15*18 #85*18 = minimum_length*18
num_classes = 4 # 

thumb_labels =  [[0] for i in range(30)]

stacked_thumb = [np.random.rand(270)for i in range(30)] 

peace_labels =  [[1] for i in range(30)]

stacked_peace =  [np.random.rand(270)for i in range(30)] 

five_fingers_labels = [[2] for i in range(30)] 

stacked_five_fingers =  [np.random.rand(270)for i in range(30)] 

no_pinky_labels = [[3] for i in range(30)]

stacked_no_pinky =  [np.random.rand(270)for i in range(30)] 

###Training Dataset
labels = np.vstack((thumb_labels[0:30], peace_labels[0:30], 
                    five_fingers_labels[0:30], no_pinky_labels[0:30]))
labels = np.squeeze(labels)

stacked_features = np.vstack((stacked_thumb[0:30], stacked_peace[0:30], 
                              stacked_five_fingers[0:30], stacked_no_pinky[0:30]))

stacked_features = np.array(stacked_features, dtype = np.float32)

#parse the Dataset using the dataset API
dataset = tf.data.Dataset.from_tensor_slices( (stacked_features, labels)).shuffle(buffer_size = 40000).repeat()
dataset = dataset.batch(batch_size)
itera = dataset.make_one_shot_iterator()

def neural_net(x):
    layer1 = tf.layers.dense(x, n_hidden_1,activation=tf.nn.relu, name="layer1")
    layer2 = tf.layers.dense(layer1, n_hidden_2,activation=tf.nn.relu,name="layer2")
    layer3 = tf.layers.dense(layer2, n_hidden_3,activation = tf.nn.relu, name="layer3")
    fclayer = tf.layers.dense(layer3,4, name="fclayer")
    return fclayer

x_batch, y_batch = itera.get_next()

# Construct model
logits = neural_net(x_batch)


# Define loss and optimizer
loss_op = tf.reduce_mean(tf.nn.sparse_softmax_cross_entropy_with_logits(logits=logits, labels=y_batch))
optimizer = tf.train.AdagradOptimizer(learning_rate=learning_rate, initial_accumulator_value=0.1)
train_op = optimizer.minimize(loss_op)

# Evaluate model
val = tf.argmax(logits)
correct_pred = tf.equal(tf.argmax(logits,1), y_batch)
accuracy = tf.reduce_mean(tf.cast(correct_pred, tf.float32))

init = tf.global_variables_initializer()
saver = tf.train.Saver()

sess = tf.Session()
sess.run(init)
for step in range(200):
    _,loss_value,acc = sess.run([train_op,loss_op,correct_pred])
    if step%25==0:
        print("iter: {}, LOss: {:.4f}".format(step,loss_value))
save_path = saver.save(sess, "./brandoning.ckpt")
tf.train.write_graph(sess.graph_def, './ckptFiles/','simple_as_binary.pbtxt')
tf_path = './simple_as_binary.pb'

tf.reset_default_graph()
saver = tf.train.Saver()`

**Comment by b6fung** [comment]
Not sure if this would help, but if there are iterators in the model would it fail the tlconverter?  I was under the impression only the weights and the shape of the graph was being saved.  I apologize for the inconvenience.

**Comment by DawerG** [comment]
The problem lies in not freezing the graph before running the converter. Conversion of training graphs is neither supported nor desired. Here is the solution. 

```
from tensorflow.python.tools import strip_unused_lib
from tensorflow.python.framework import dtypes
from tensorflow.python.platform import gfile

input_node_names = ['IteratorGetNext']
output_node_names = ['fclayer/BiasAdd']

gdef = tf.graph_util.convert_variables_to_constants(
        sess,
        sess.graph_def,
        output_node_names)
gdef = strip_unused_lib.strip_unused(
        input_graph_def = gdef,
        input_node_names = input_node_names,
        output_node_names = output_node_names,
        placeholder_type_enum = dtypes.float32.as_datatype_enum)
# Save it to an output file
frozen_model_file = './Issue#233.pb'
with gfile.GFile(frozen_model_file, "wb") as f:
    f.write(gdef.SerializeToString())
```

```
import tfcoreml
# Supply a dictionary of input tensors' name and shape (with # batch axis)
input_tensor_shapes = {"IteratorGetNext:0":[1,270]} # batch size is 1

coreml_model_file = './Issue#233.mlmodel'  # Output CoreML model path
output_tensor_names = ['fclayer/BiasAdd:0'] # The TF model's ouput tensor name

coreml_model = tfcoreml.convert(
        tf_model_path='./Issue#233.pb',
        mlmodel_path=coreml_model_file,
        input_name_shape_dict=input_tensor_shapes,
        output_feature_names=output_tensor_names)
```

For further reference, please checkout the python notebooks in the example directory. Closing the issue for now.
